tqdm/_tqdm.py

```diff
@@ -752,10 +763,9 @@ def __init__(self, iterable=None, desc=None, total=None, leave=True,
         self.start_t = self.last_print_t
 
     def __len__(self):
-        return self.total if self.iterable is None else \
-                (self.iterable.shape[0] if hasattr(self.iterable, "shape")
-                 else len(self.iterable) if hasattr(self.iterable, "__len__")
-                 else self.total)
+        return (self.iterable.shape[0] if hasattr(self.iterable, 'shape')
+                else len(self.iterable)) if self.iterable is not None \
```

Comment:
To be honest, I don't remember why I removed it. I only know that I've been making changes on tqdm 4.7.4 and then transfering them on 4.11.2.